src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -647,62 +722,63 @@ internal async void ExecutePackageDownload(string name, PackageVersion package,
 
                 if (uninstallRequiringUserModifications.Any())
                 {
-                    MessageBoxService.Show(String.Format(Resources.MessageUninstallToContinue,
-                        DynamoViewModel.BrandingResourceProvider.ProductName,
-                        JoinPackageNames(uninstallRequiringUserModifications)),
-                        Resources.CannotDownloadPackageMessageBoxTitle,
-                        MessageBoxButton.OK, MessageBoxImage.Error);
-                    return;
+                    var conflictingPkgs = JoinPackageNames(uninstallRequiringUserModifications);
+                    var message = string.Format(Resources.MessageForceInstallOrUninstallToContinue, packageToDownload, conflictingPkgs,
+                        DynamoViewModel.BrandingResourceProvider.ProductName);
+                    
+                    var dialogResult = MessageBoxService.Show(message,
+                        Resources.PackagesInUseConflictMessageBoxTitle,
+                        MessageBoxButton.YesNo, MessageBoxImage.Exclamation);
+
+                    if(dialogResult == MessageBoxResult.No) return;
                 }
 
                 var settings = DynamoViewModel.Model.PreferenceSettings;
-
                 if (uninstallsRequiringRestart.Any())
                 {
-                    var message = string.Format(Resources.MessageUninstallToContinue2,
-                        DynamoViewModel.BrandingResourceProvider.ProductName,
-                        JoinPackageNames(uninstallsRequiringRestart),
-                        name + " " + package.version);
-                    // different message for the case that the user is
-                    // trying to install the same package/version they already have installed.
-                    if (uninstallsRequiringRestart.Count == 1 &&
-                        uninstallsRequiringRestart.First().Name == name &&
-                        uninstallsRequiringRestart.First().VersionName == package.version)
-                    {
-                        message = String.Format(Resources.MessageUninstallSamePackage, name + " " + package.version);
-                    }
+                    var conflictingPkgs = JoinPackageNames(uninstallsRequiringRestart);
+                    var message = string.Format(Resources.MessageForceInstallOrUninstallUponRestart, packageToDownload,
+                        conflictingPkgs, DynamoViewModel.BrandingResourceProvider.ProductName);
+
                     var dialogResult = MessageBoxService.Show(message,
-                        Resources.CannotDownloadPackageMessageBoxTitle,
-                        MessageBoxButton.YesNo, MessageBoxImage.Error);
+                        Resources.LoadedPackagesConflictMessageBoxTitle,
+                        MessageBoxButton.YesNoCancel, MessageBoxImage.Exclamation);
 
-                    if (dialogResult == MessageBoxResult.Yes)
+                    if (dialogResult == MessageBoxResult.No)
```

Comment:
The `YesNoCancel` message's question here is 
`Do you wish to continue installing it while keeping the packages that are already installed?`
YEs -> obviously means installing and keeping conflicts loaded too.
No -> Not so obvious to me that it means marking the conflict as Uninstalled....
Cancel-> Obvious it should cancel the whole operation.....But I do not see it being handled here

Reply:
I guess I missed this part `If not, {2} needs to uninstall {1} to continue. Restart {2}, then download {0} again.`
Good enough for me